### PR TITLE
guide to install zimbra x web

### PR DESCRIPTION
**Pull request by silentsakky**

### CODE REVIEW

**Comment by silentsakky** [comment]
https://github.com/Zimbra/zm-x-web/wiki/Customising-Zimbra-X-Webclient

**Comment by gordyt** [comment]
@silentsakky I'll take a look at this when I get a chance.  Will need to check that PR branch out locally, generate the three HTML exports, verify no warnings and verify that formatting looks good. I'm swamped ATM.

File: zimbra-x-webclient.adoc [hunk 1]
```diff
@@ -0,0 +1,93 @@
+= Installing {product-short} X Webclient
+
+Optionally you can install {product-short} X Webclient, which is currently in beta stage and you can provide early feedback to {product-short}.
+Below are the steps to setup {product-short} X Webclient on {product-short}
+
+. Run {product-short} X Webclient on any of the mailbox server
```
Comment (billneff79) [inline zimbra-x-webclient.adoc]: `servers` plural

File: zimbra-x-webclient.adoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	* Make sure latest version of **NodeJS** and **NPM** is installed
+	* In case of multinode environment make sure you are selecting a node which is currently working as a mailbox node
+	* Clone respository of {product-short} X Webclient by executing `git clone https://github.com/Zimbra/zm-x-web.git`
+	* If you want to do any customizations on top of {product-short} X Webclient then you can do as shown here https://github.com/Zimbra/zm-x-web/wiki/Customising-Zimbra-X-Webclient[Customising-Zimbra-X-Webclient]
+	* Install npm modules by executing `npm install`
+	* Create a production build by executing `npm run build`
```
Comment (billneff79) [inline zimbra-x-webclient.adoc]: If they customized a folder other than `clients/default` they will need to specify `CLIENT=foo` or whatever their client folder is when they build

File: zimbra-x-webclient.adoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+	* add below block as an extra upstream block, this should contain hostname of mailbox node which contains zm-x-web, make sure to give correct port which was used when executing npm run serve command above
+
+	upstream zimbra_x_webclient
+	{
+		server    <hostname_of_mailbox_which_hosts_zimbra_x_webclient>:9090 fail_timeout=10s version=8.8.8_GA_1231;
+		zmauth;
+	}
+	
+	* Navigate to `conf/nginx/includes/` to directory
```
Comment (billneff79) [inline zimbra-x-webclient.adoc]: Grammar: "Navigate to the `confi/nginx/includes/` directory"

File: zimbra-x-webclient.adoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	* Copy `nginx.conf.web.admin.default` to same directory and name it as `nginx.conf.web.zimbrax.default`
+	* Remove all location blocks from `nginx.conf.web.zimbrax.default` file, and add location blocks shown below
+	* After this restart proxy by `zmproxyctl restart`
+	* Check if {product-short} X Webclient is working on https://<hostname_of_proxy>:9090/
```
Comment (billneff79) [inline zimbra-x-webclient.adoc]: So this makes it available on a specific port, not at a given subdomain or sub-path?

**Review by billneff79 — CHANGES_REQUESTED** [review]
small changes needed

File: zimbra-x-webclient.adoc (hunk 1 above)
Comment (silentsakky) [reply zimbra-x-webclient.adoc]: Fixed

File: zimbra-x-webclient.adoc (hunk 2 above)
Comment (silentsakky) [reply zimbra-x-webclient.adoc]: Fixed

File: zimbra-x-webclient.adoc (hunk 3 above)
Comment (silentsakky) [reply zimbra-x-webclient.adoc]: Fixed

File: zimbra-x-webclient.adoc (hunk 4 above)
Comment (silentsakky) [reply zimbra-x-webclient.adoc]: Updated doc to have configuration for subdomain also
sub-path is not possible as it is conflicting with configuration of legacy webclient

**Review by billneff79 — APPROVED** [review]
Looks good.  For testing, we should have someone else use these instructions to create a setup and give any additional feedback

**Review by jiteshsojitra — APPROVED** [review]

File: zimbra-x-webclient.adoc [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+	* Make sure latest version of **NodeJS** and **NPM** is installed
+	* In case of multinode environment make sure you are selecting a node which is currently working as a mailbox node
+	* Clone respository of {product-short} X Webclient by executing `git clone https://github.com/Zimbra/zm-x-web.git`
+	* If you want to do any customizations on top of {product-short} X Webclient then you can do as shown here https://github.com/Zimbra/zm-x-web/wiki/Customising-Zimbra-X-Webclient[Customising-Zimbra-X-Webclient]
+	* Install npm modules by executing `npm install`
+	* Create a production build by executing `npm run build`.  If customisations were made in a client directory other that `default`, e.g. in the `client/foo` directory, then specify the client directory name as an environment variable on the command line like `CLIENT=foo npm run build`
+	* `PORT=9090 npm run serve` (We are specifying port 9090 to make sure it doesn't conflict with other used ports in {product-short})
+
+. Configure nginx to route requests properly
```
Comment (miteshsavani) [inline zimbra-x-webclient.adoc]: Add server name 'Configure nginx to route requests properly **on proxy server**'

File: zimbra-x-webclient.adoc [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	* add below line in `conf/nginx/templates/nginx.conf.web.template` file at the end of file where other files are getting included
```
Comment (miteshsavani) [inline zimbra-x-webclient.adoc]: Mention root path /opt/zimbra/  in location `conf/nginx/templates/nginx.conf.web.template`

File: zimbra-x-webclient.adoc [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+	* add below block as an extra upstream block, this should contain hostname of mailbox node which contains zm-x-web, make sure to give correct port which was used when executing npm run serve command above
+
+	upstream zimbra_x_webclient
+	{
+		server    <hostname_of_mailbox_which_hosts_zimbra_x_webclient>:9090 fail_timeout=10s version=8.8.8_GA_1231;
+		zmauth;
+	}
+	
+	* Navigate to the `conf/nginx/includes/` directory
```
Comment (miteshsavani) [inline zimbra-x-webclient.adoc]: Mention root path /opt/zimbra/  in location 'conf/nginx/includes/

File: zimbra-x-webclient.adoc [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+		- Update port number in listen block on second line of server block to `9090` (or any selected port)
+		- Restart proxy by `zmproxyctl restart`
+		- Check if {product-short} X Webclient is working on https://<hostname_of_proxy>:9090/
+
+	* Use subdomain to access {product-short} X Webclient
+		- Update port number in listen block on second line of server block to `443`, this will make sure subdomain is listening on default ssl port
+		- add `server_name` directive under listen directive and specify subdomain which we need to use to listen to requests `server_name     <subdomain>.<hostname_of_proxy>`
+		- Restart proxy by `zmproxyctl restart`
+		- Check if {product-short} X Webclient is working on https://<subdomain>.<hostname_of_proxy>/
+
```
Comment (miteshsavani) [inline zimbra-x-webclient.adoc:46]: Put the Notes like 'User should have to access Zimbra X through Proxy '

File: zimbra-x-webclient.adoc (hunk 5 above)
Comment (silentsakky) [reply zimbra-x-webclient.adoc]: Fixed

File: zimbra-x-webclient.adoc (hunk 6 above)
Comment (silentsakky) [reply zimbra-x-webclient.adoc]: Fixed

File: zimbra-x-webclient.adoc (hunk 7 above)
Comment (silentsakky) [reply zimbra-x-webclient.adoc]: Fixed

File: zimbra-x-webclient.adoc (hunk 8 above)
Comment (silentsakky) [reply zimbra-x-webclient.adoc:46]: Fixed

**Review by miteshsavani — APPROVED** [review]

**Comment by gordyt** [comment]
Generated multi service guide and did a quick visual inspection to ensure formatting looked OK. Merged pull request branch into develop and pushed to origin. Deleted pull request branch from origin.

**Comment by silentsakky** [comment]
Thanks @gordyt for taking care of this

**Pull request merged**